tree/treeplayer/src/TSimpleAnalysis.cxx

```diff
@@ -0,0 +1,384 @@
+/// Handle the expression lines of the input file in order to pass the
+/// elements to the members of the object.
+///
+/// param[in] line line or read from the input file either the expression passed to the constructor
+
+std::string TSimpleAnalysis::HandleExpressionConfig(const std::string& line)
+{
+   static const std::string kCutIntr = " if ";
+
+   std::size_t equal = line.find("=");
+   if (equal == std::string::npos)
+      return "Error: missing '='";
+
+   // Set the histName value
+   std::string histName = line.substr(0, equal);
+   DeleteSpaces(histName);
+   if (histName.empty())
+      return "Error: no histName found";
+
+   //Set the histExpression value
+   std::size_t cutPos = line.find(kCutIntr, equal);
+   std::string histExpression = line.substr(equal + 1, cutPos - equal - 1);
+   DeleteSpaces(histExpression);
+   if (histExpression.empty())
+      return "Error: no expression found";
+
+   // Set the histCut value
+   std::string histCut;
+   if (cutPos != std::string::npos) {
+      histCut = line.substr(cutPos + kCutIntr.size());
+      DeleteSpaces(histCut);
+      if (histCut.empty())
+         return "Error: invalid syntax";
+   }
+   else
+      histCut = "";
+
+   // Set the map that contains the histName, histExpressions and histCut values
+   auto check = fHists.insert(std::make_pair((const std::string&)histName,
+                                             std::make_pair(histExpression, histCut)));
+
+   // Check if there are histograms with the same name
+   if (!check.second)
+      return "Duplicate histogram name";
+   return "";
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Constructor for the case of command line parsing arguments. It sets the members
+/// of the object.
+///
+/// \param[in] output name of the output file
+/// \param[in] inputFiles name of the input .root files
+/// \param[in] expressions what is showed in the histograms
+/// \param[in] treeName name of the tree
+/// \throws std::runtime_error in case of ill-formed expressions
+
+TSimpleAnalysis::TSimpleAnalysis(const std::string& output,
+                                 const std::vector<std::string>& inputFiles,
+                                 const std::vector<std::string>& expressions,
+                                 const std::string& treeName = ""):
+   fInputFiles(inputFiles), fOutputFile(output), fTreeName(treeName)
+{
+   for (const std::string& expr: expressions) {
+      std::string errMessage = HandleExpressionConfig(expr);
+      if (!errMessage.empty())
+         throw std::runtime_error(errMessage + " in " +  expr);
+   }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Extract the name of the tree from the first input file when the tree name
+/// isn't in the configuration file. Returns the name of the tree.
+
+std::string TSimpleAnalysis::ExtractTreeName()
```

Comment:
Can you make this static and non-member, taking the name of the first input file.